util/dump-ledger/src/lib.rs

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! A utility to dump a ledger's contents as JSON.
+
+#![deny(missing_docs)]
+
+mod error;
+
+use clap::Parser;
+use mc_blockchain_types::BlockIndex;
+use mc_ledger_db::Ledger;
+use serde_json::to_string_pretty as to_json;
+
+pub use error::{Error, Result};
+
+/// Parameters for [dump_ledger].
+#[derive(Debug, Clone, Parser)]
+pub struct DumpParams {
+    /// Optional first index; defaults to 0.
+    #[clap(long, short, env = "MC_FIRST_INDEX")]
+    pub first_index: Option<BlockIndex>,
```

Comment:
IFF done would require further code and test changes
```suggestion
    /// Optional first index.
    #[clap(long, short, default_value = "0", env = "MC_FIRST_INDEX")]
    pub first_index: BlockIndex,
```